### DISCUSSION TITLE
Shortened .m4a files

### DISCUSSION THREAD

**Discussion by ghender3** [mifi/lossless-cut#1751] [Q&A]
If I try to export an .m4a audio file _without_  making any selections or cuts, the exported file always has a slightly shorter duration than the original. In MediaInfo, the exported file mentions source duration, and actual duration. Similar for frame count - source and actual. On a file I must tried, the original is 2:42.215. The export is always 2:42.167, regardless of different things I try in the export dialog that appears just before the export.

If I recall correctly, the destination file frame count is always either 2 or 3 frames less than the original source.

I am using the current version, and I'm making sure that I'm deselecting anything that isn't show as an audio track.

This isn't a huge problem, but it's something I've seen in LosslessCut for about 3 years now. Why can't it simply export an .m4a without altering it?

**Comment by mifi** [comment]
that' odd. have you tried with the ffmpeg command line and see if the same thing happens?
```bash
ffmpeg -i input -c copy out.mp4
```

if so, then it's a bug or issue in ffmpeg, and not much losslesscut can do

**Comment by ghender3** [comment]
I just tried ffmpeg in Terminal. Same issue - a song I tried is 2:29.420 before ffmpeg copy operation, 2:29.373 after. Frame count went from 6435 to 6433.

I understand that Losslesscut is a very handy GUI for ffmpeg, so you’re right. 

Just FYI, mp4box (maintained by GPAC) and GUIs for mp4box don’t truncate files like this. But the GUIs are pretty rudimentary and not well implemented.

Thanks

Gordon Henderson

**Reply by mifi** [reply]
Ok thanks for the research. Feel free to report an issue at ffmpeg if you believe it’s a bug.

**Comment by ghender3** [comment]
I think I’ve found out what is probably at the root of this issue with ffmpeg dropping frames, samples and shortening durations with .m4a audio. Ffmpeg apparently doesn’t “respect” variable bit rate audio as being such, at least not when using the below command. In every case, whether using LosslessCut or ffmpeg itself, the original audio shows in MediaInfo as VBR. After copying, it’s always CBR audio. I just tried using LosslessCut on a piece of audio that I converted to .m4a in iTunes using CBR mode, and there are NOT 2 missing frames, and the duration only differs by 1 millisecond.  I’ve seen dozens of bug reports pertaining to VBR audio on FFMPEG’s bug tracker site, so I wouldn’t think adding yet another report would contribute anything. What I don’t know is whether there is an ffmpeg command that would force it to “respect” the VBR flag.

Gordon Henderson
